File: algorithms/2_medium/W/02079_watering_plants/solution.py
class Solution:
    def wateringPlants(self, plants: List[int], capacity: int) -> int:
        can = capacity
        count = 0
        plants_len = len(plants)

        for i in range(plants_len):
            can -= plants[i]
            count += 1
            if i < plants_len - 1 and can < plants[i + 1]:
                count += 2 * (i + 1)
                can = capacity

        return count

File: algorithms/2_medium/W/02079_watering_plants/solution.cpp
class Solution {
public:
    int wateringPlants(vector<int>& plants, int capacity) {
        int can{capacity};
        int count{0};
        const size_t len{plants.size()};

        for (size_t i{0}; i < len; ++i) {
            can -= plants[i];
            count += 1;
    
            if (i < len - 1 && can < plants[i + 1]) {
                count += 2 * (i + 1);
                can = capacity;
            }
        }

        return count;
    }
};
